### DISCUSSION TITLE
Asynchronous reprojection support

### DISCUSSION THREAD

**Discussion by ghost** [godotengine/godot-proposals#5892] [3D]
Hi, I've watched [Linus's video](https://www.youtube.com/watch?v=IvqrlgKuowE) on it and tried the [Unity demo](https://www.youtube.com/watch?v=VvFyOFacljg) and I think it is great. However, in the comments the author states that one should have game engine support for it to work on real low frame rates. I would like this technology to become standard for non-VR games as well so that's why I'm creating this discussion. What do you think about it?

**Comment by Calinou** [comment]
See https://www.reddit.com/r/godot/comments/zakw3l/asynchronous_reprojection/ and https://github.com/Calinou/godot-reprojection-demo.

**Reply by ghost** [reply]
Should I create an issue then?

**Reply by Calinou** [reply]
> Should I create an issue then?

No, as we'd need a concrete technical implementation that can effectively avoid the big issue with this proof-of-concept: it needs to be able to render frames even when the process is not actually putting enough frames per second. This is a hard problem to resolve in any game engine; it seems that only driver intervention can do this well (like in DLSS Frame Generation).

Without a way to render frames even when the GPU is fully used, having asynchronous reprojection in core wouldn't be useful.

I'd also wait until the next iteration of FSR is released, as it may feature a form of frame generation (not confirmed by AMD, but the promised performance gains of FSR 3 don't leave much room for alternatives  here).

**Reply by ghost** [reply]
Got it, thanks
